backend/rules/rules.js

```diff
@@ -0,0 +1,119 @@
+const troopsRule = {
```

Comment:
```suggestion
const buildingsConstructionCost: 100;
const buildingsConstructionTimeInSeconds: 600;

const troopsConstructionCost: 100;
const troopsConstructionTimeInSeconds: 60;
const troopsRule = {
```

Reply:
this is just a suggestion, you can also use them inside the objects, whatever is more readable for you, but don't write them into every level.
also use more descriptive names, because `buildSecond` is not so obvious for someone who doesn't know the application. You can say something like `buildingConstructionTimeInSeconds`, or `...Generation...`, `...Creation...`, etc... 
Same for `upgradeSecond` (`upgradingTimeInSeconds` or anything similar)